core/src/main/scala/anorm/package.scala

```diff
@@ -33,15 +35,30 @@ package object anorm {
   }
 
   /** Structural type for Oracle timestamp wrapper. */
-  type TimestampWrapperOracle = { def timestampValue: java.sql.Timestamp }
+  type OracleTimestampWrapper = { def timestampValue: java.sql.Timestamp }
 
-  object TimestampWrapperOracle {
+  object OracleTimestampWrapper {
     import scala.language.reflectiveCalls
 
     def unapply(that: Any): Option[java.sql.Timestamp] = try {
-      Some(that.asInstanceOf[TimestampWrapperOracle].timestampValue)
+      Some(that.asInstanceOf[OracleTimestampWrapper].timestampValue)
+    } catch {
+      case _: NoSuchMethodException => None
+      case _: SQLException => None
+    }
+  }
+
+  /** Structural type for Oracle string wrapper. */
+  type OracleStringWrapper = { def stringValue: String }
```

Comment:
Prefer a more neutral naming, as the structural type can be used for any value providing the `stringValue`